spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkPartitioningAwareScan.java

```diff
@@ -232,6 +232,16 @@ protected synchronized List<ScanTaskGroup<T>> taskGroups() {
     return taskGroups;
   }
 
+  private long targetSplitSize() {
```

Comment:
Don't we need this in other places as well? Or does everything go through `SparkPartiitoningAwareScan` now?

Reply:
I'll check but I think so.

Reply:
The only place we miss is `SparkChangelogScan` as it plans tasks directly. We can update it later to plan files first or it will come automatically once we support adaptive split size in core.